### PR TITLE
feat(performance-detector-thresholds): Added new endpoint for admin s…

### PR DESCRIPTION
**Pull request by Abdkhan14**
First step of splitting project detector threshold settings to two key value pairs. One representing only the admin settings(enabling/disabling) and the other representing the customer configurable thresholds. More [context](https://www.notion.so/sentry/Detector-Threshold-Configuration-f8cb07e7cceb42388cedb09ea05fc116?pvs=4#b6cd5d2191324058955cf484f5342cca).

- Added a new endpoint that allows superusers to add enabling/disabling of detection.
- Added tests.

### CODE REVIEW

File: src/sentry/api/endpoints/project_performance_issue_settings_admin.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    db_on_main_thread_detection_enabled = serializers.BooleanField(required=False)
+    file_io_on_main_thread_detection_enabled = serializers.BooleanField(required=False)
+    consecutive_db_queries_detection_enabled = serializers.BooleanField(required=False)
+    large_render_blocking_asset_detection_enabled = serializers.BooleanField(required=False)
+    slow_db_queries_detection_enabled = serializers.BooleanField(required=False)
+
+
+@region_silo_endpoint
+class ProjectPerformanceIssueAdminSettingsEndpoint(ProjectEndpoint):
+    permission_classes = (SuperuserPermission,)  # type: ignore
```
Comment (sentaur-athena) [inline src/sentry/api/endpoints/project_performance_issue_settings_admin.py:29]: Can you please explain why the permission is limited to superuser? From the endpoint name sounds like something we want to open to org admins/owners?

File: src/sentry/api/urls.py [hunk 2]
```diff
@@ -2256,6 +2259,11 @@
         ProjectPerformanceIssueSettingsEndpoint.as_view(),
         name="sentry-api-0-project-performance-issue-settings",
     ),
+    url(
+        r"^(?P<organization_slug>[^\/]+)/(?P<project_slug>[^\/]+)/performance-issues/configure/admin/$",
```
Comment (sentaur-athena) [inline src/sentry/api/urls.py:2263]: The url ending with admin resource means that this endpoint is adding admin on put, while from the serializer I can see it's updating a bunch of configs. Can you explain what you're trying to do with this vs /configure?

**Review by sentaur-athena — CHANGES_REQUESTED** [review]
Sending back to your queue for more clarification.

File: src/sentry/api/endpoints/project_performance_issue_settings_admin.py (hunk 1 above)
Comment (schew2381) [reply src/sentry/api/endpoints/project_performance_issue_settings_admin.py:29]: Adding onto this I see `admin` and `superuser` seem to be used interchangeably (eg: `ProjectPerformanceIssueAdminSettingsEndpoint` class name w/ superuser permission reqs). We are currently working in the process of setting API design standards, and some of those standards include:
- having the be URL representative of the resource hierarchy
- having an API do only one thing that can be guessed intuitively from url

If the endpoint is intended to be used for superusers only, would it be possible to replace references to admin with superuser in the file name, class names, and URL to reflect that?

**Comment by Abdkhan14** [comment]
Hey @sentaur-athena! This PR was for my [project](https://www.notion.so/sentry/Detector-Threshold-Configuration-f8cb07e7cceb42388cedb09ea05fc116#b6cd5d2191324058955cf484f5342cca) that'll allow users to alter the detector thresholds and limits the ability to enable/disable detection to sentry superusers. This is a failed approach,  we won't be needing this endpoint. I'm closing this PR. Sorry for the trouble.

**Pull request closed without merging**